kornia/color/adjust.py

```diff
@@ -225,6 +227,169 @@ def adjust_brightness(input: torch.Tensor,
+            res.append(_posterize_one(input[:, i], bits[i]))
+        return torch.stack(res, dim=1)
+
+    assert bits.shape == input.shape[:len(bits.shape)], \
+        f"Batch and channel must be equal between bits and input. Got {bits.shape}, {input.shape[:len(bits.shape)]}."
+    _input = input.view(-1, *input.shape[len(bits.shape):])
+    _bits = bits.flatten()
+    for i in range(input.shape[0]):
+        res.append(_posterize_one(_input[i], _bits[i]))
+    res = torch.stack(res, dim=0)
+    return res.reshape(*input.shape)
+
+
+def sharpness(input: torch.Tensor, factor: Union[float, torch.Tensor]) -> torch.Tensor:
+    r"""Implements Sharpness function from PIL using torch ops.
+    Args:
+        input (torch.Tensor): image or batched images for sharpness.
+        bits (float or torch.Tensor): factor of sharpness strength. Must be above 0.
+            If float or one element tensor, input will be sharpened by the same factor across the whole batch.
+            If 1-d tensor, input will be sharpened element-wisely, len(factor) == len(input).
+    Returns:
+        torch.Tensor: Sharpened image or images.
+    """
+    input = to_bchw(input)
+    if isinstance(factor, torch.Tensor):
+        factor = factor.squeeze()
+        if len(factor) != 0:
+            assert input.size(0) == factor.size(0), \
+                f"Input batch size shall match with factor size if 1d array. Got {input.size(0)} and {factor.size(0)}"
+    else:
+        factor = float(factor)
+    kernel = torch.tensor([
+        [1, 1, 1],
+        [1, 5, 1],
+        [1, 1, 1]
+    ], dtype=input.dtype).view(1, 1, 3, 3).repeat(3, 1, 1, 1)
+
+    # This shall be equivalent to depthwise conv2d:
+    # Ref: https://discuss.pytorch.org/t/depthwise-and-separable-convolutions-in-pytorch/7315/2
+    degenerate = torch.nn.functional.conv2d(input, kernel, bias=None, stride=1, groups=input.size(1))
+    degenerate = torch.clamp(degenerate, 0., 1.)
+
+    mask = torch.ones_like(degenerate)
+    padded_mask = torch.nn.functional.pad(mask, [1, 1, 1, 1])
+    padded_degenerate = torch.nn.functional.pad(degenerate, [1, 1, 1, 1])
+    result = torch.where(padded_mask == 1, padded_degenerate, input)
+
+    def _blend_one(input1: torch.Tensor, input2: torch.Tensor, factor: Union[float, torch.Tensor]) -> torch.Tensor:
+        if isinstance(factor, torch.Tensor):
+            factor = factor.squeeze()
+            assert len(factor) == 0, f"Factor shall be a float or single element tensor. Got {factor}"
+        if factor == 0.:
+            return input1
+        if factor == 1.:
+            return input2
+        diff = (input2 - input1) * factor
+        res = input1 + diff
+        if factor > 0. and factor < 1.:
+            return res
+        return torch.clamp(res, 0, 1)
+    if isinstance(factor, (float)) or len(factor) == 0:
+        return _blend_one(input, result, factor)
+    return torch.stack([_blend_one(input[i], result[i], factor[i]) for i in range(len(factor))])
```

Comment:
For sharpness I found this much clear to me
https://stackoverflow.com/questions/4993082/how-to-sharpen-an-image-in-opencv

We support gaussian blur and add weighted

Maybe we could support both implementations by adding a flag `method` to the function signature. 

```python
if method == 'pillow':
  return _sharpen_pil(...)
elif method == 'default':
   return _sharpen(...)
else:
   raise ERROR
```

This could be applied to the other methods with different possible implementations. However we should find what are the actual methods names not names of the libraries implementing. /cc @ducha_aiki @krrish94

Reply:
I think it is better to stick with one method. Or we shall support multi-backends globally but in a decent manner somehow, not rely on if else in each method.

Reply:
it's quite common in computer vision libraries to have different methods solving same problem/task.

Reply:
I mean it is better to have a global switch like ```kornia.backend = 'PIL'``` or ```kornia.backend = 'OpenCV'``` for a global behaviour control as well.